nautobot/docs/development/apps/migration/ui-component-framework.md

```diff
@@ -0,0 +1,244 @@
+# Migrating to UI Component Framework
+A guide for app developers transitioning to the UI Component Framework.
+
+## Introduction
+
+This guide helps you migrate your existing Nautobot app views to use the new UI Framework. The framework provides a declarative approach to building object detail views, reducing boilerplate code while ensuring consistency across the platform.
+
+For complete UI Framework documentation, see: [Nautobot UI Framework Documentation](../../core/ui-component-framework.md)
+
+## Why Migrate?
+
+### Benefits
+- Reduced template maintenance
+- Consistent UI patterns across apps
+- Built-in responsive layouts
+- Standardized component behavior
+
+### Before and After Example
+
+Before (Template-based):
+```python
+# views.py
+class DeviceDetailView(generic.ObjectView):
+    queryset = Device.objects.all()
+    template_name = 'myapp/device_detail.html'
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context['related_devices'] = self.object.related_devices.all()
+        context['custom_fields'] = self.object.get_custom_fields()
+        return context
+
+# template.html
+{% extends 'base.html' %}
+{% block content %}
+<div class="row">
+    <div class="col-md-6">
+        <div class="panel">
+            <!-- Manual HTML for device fields -->
+        </div>
+    </div>
+    <div class="col-md-6">
+        <!-- More manual HTML -->
+    </div>
+</div>
+{% endblock %}
+```
+
+After (UI Framework):
+```python
+from nautobot.core.views import ObjectView
+from nautobot.core.views.content import ObjectDetailContent
+from nautobot.core.views.panels import (
+    ObjectFieldsPanel,
+    ObjectsTablePanel,
+    TextPanel,
+)
```

Comment:
App should import from `nautobot.apps.ui` instead; also it's a bit confusing/misleading that the "After" includes these imports but the "before" doesn't include any imports?